polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -112,20 +112,36 @@ class ValueOrUnsigned {
   }
 
 private:
-  template <typename T,
-            typename = std::enable_if_t<llvm::is_one_of<
-                T, arith::AddIOp, arith::MulIOp, arith::MaxSIOp>::value>>
+  template <typename T> struct is_supported_binop {
```

Comment:
Honestly having more templates makes the code longer and harder to read. I prefer the old implementation, fairly concise and shorter. WDYT?

Reply:
Discussed offline with Victor and Ettore, we agreed to keep the code like it was.